src/pool/srv_pool.c

```diff
@@ -3063,6 +3143,72 @@ ds_pool_svc_get_prop(uuid_t pool_uuid, d_rank_list_t *ranks,
 	return rc;
 }
 
+int
+ds_pool_add(uuid_t pool_uuid, int ntargets, uuid_t target_uuids[],
+		const d_rank_list_t *rank_list, int ndomains,
+		const int *domains, d_rank_list_t *svc_ranks)
+{
+	int				rc;
+	struct rsvc_client		client;
+	crt_endpoint_t			ep;
+	struct dss_module_info		*info = dss_get_module_info();
+	crt_rpc_t			*rpc;
+	struct pool_extend_in		*in;
+	struct pool_extend_out		*out;
+
+	rc = rsvc_client_init(&client, svc_ranks);
+	if (rc != 0)
+		return rc;
+
+rechoose:
+
+	ep.ep_grp = NULL; /* primary group */
+	rsvc_client_choose(&client, &ep);
+
+	rc = pool_req_create(info->dmi_ctx, &ep, POOL_EXTEND, &rpc);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to create pool extend rpc: "
+			""DF_RC"\n", DP_UUID(pool_uuid), DP_RC(rc));
+		D_GOTO(out_client, rc);
+	}
+
+	in = crt_req_get(rpc);
+	uuid_copy(in->pei_op.pi_uuid, pool_uuid);
+	in->pei_ntgts = ntargets;
+	in->pei_ndomains = ndomains;
+	in->pei_tgt_uuids.ca_count = 1;
```

Comment:
Not sure I follow, why this is 1?

Reply:
You are correct, fixed in most recent patch.